src/reducers-actions/localTokens.ts

```diff
@@ -0,0 +1,93 @@
+import { Actions } from 'reducers-actions'
+import { STORAGE_KEY_DISABLED_TOKENS_ADDRESSES } from 'const'
+
+export interface LocalTokensState {
+  disabled: Set<string>
+}
+
+export const enum ActionTypes {
+  ENABLE_TOKEN = 'ENABLE_TOKEN',
+  DISABLE_TOKEN = 'DISABLE_TOKEN',
+  UPDATE_ALL_DISABLED = 'UPDATE_ALL_DISABLED',
+}
+
+interface TokenAddressPayload {
+  tokenAddress: string
+}
+
+type EnableLocalToken = Actions<ActionTypes.ENABLE_TOKEN, TokenAddressPayload>
+type DisableLocalToken = Actions<ActionTypes.DISABLE_TOKEN, TokenAddressPayload>
+type UpdateAllDisabled = Actions<ActionTypes.UPDATE_ALL_DISABLED, Partial<LocalTokensState>>
+
+type ReducerActionType = EnableLocalToken | DisableLocalToken | UpdateAllDisabled
+
+export const enableToken = (payload: TokenAddressPayload): EnableLocalToken => ({
+  type: ActionTypes.ENABLE_TOKEN,
+  payload,
+})
+export const disableToken = (payload: TokenAddressPayload): DisableLocalToken => ({
+  type: ActionTypes.DISABLE_TOKEN,
+  payload,
+})
+export const updateLocalTokens = (payload: Partial<LocalTokensState>): UpdateAllDisabled => ({
+  type: ActionTypes.UPDATE_ALL_DISABLED,
+  payload,
+})
+export const saveDisabledToStorage = ({ disabled }: LocalTokensState): void => {
+  const disabledAddressesArray = Array.from(disabled)
+  localStorage.setItem(STORAGE_KEY_DISABLED_TOKENS_ADDRESSES, JSON.stringify(disabledAddressesArray))
+}
+
+const initializeDisabled = (): Set<string> => {
+  const disabledTokensFromStorage = localStorage.getItem(STORAGE_KEY_DISABLED_TOKENS_ADDRESSES)
+  const initialDisabledSet = disabledTokensFromStorage
+    ? new Set<string>(JSON.parse(disabledTokensFromStorage))
+    : new Set<string>()
+
+  return initialDisabledSet
+}
+
+export const INITIAL_LOCAL_TOKENS_STATE: LocalTokensState = {
+  disabled: initializeDisabled(),
+}
+
+export function reducer(state: LocalTokensState, action: ReducerActionType): LocalTokensState {
+  const { disabled } = state
+
+  switch (action.type) {
+    case ActionTypes.ENABLE_TOKEN: {
+      const { tokenAddress } = action.payload
```

Comment:
Thanks to `enums`, this `action` is unequivocally determined to be `EnableLocalToken` and we know its payload's shape.
Same doesn't work for type union -- `action` would be `ReducerActionType = EnableLocalToken | DisableLocalToken | UpdateAllDisabled`

Reply:
TS is very smart, I like it 👍 

![image](https://user-images.githubusercontent.com/2352112/81404279-5f83f880-9135-11ea-87ee-e00cf9d52029.png)

Great use of enums. 
... however, I wonder what happens if I use strings instead of Enums 🤔
Just for fun, let me try... 

...It works!!! (disclaimer alert, I tried, saw it worked, and then wrote "let me try" to pretend to be interesting)
And it looks simpler, we just go straight into defining the actions
```ts
type EnableLocalToken = Actions<'ENABLE_TOKEN', TokenAddressPayload>
type DisableLocalToken = Actions<'DISABLE_TOKEN', TokenAddressPayload>
type UpdateAllDisabled = Actions<'UPDATE_ALL_DISABLED', Partial<LocalTokensState>>
```

![image](https://user-images.githubusercontent.com/2352112/81405144-627fe880-9137-11ea-90ab-2e91f5e10855.png)


Although I love enums in any programming language, I don't see the point in TS yet.

Reply:
> I don't see the point in TS yet

Why? They are not much different than in any other language

Reply:
I think what he means is that with the types, there's really no need for enums.

Reply:
Exactly, I just feel is always more verbose, with the same level of type safety, the auto-suggest is worse  --> cause for enums I need to import and then I get the autosuggest (opposed to getting it right away with string approach)